tests/test_invoices.py

```diff
@@ -0,0 +1,386 @@
+    # x (standard, stndard light, day pass)
+    # x (student, personal, company)
+
+    # fare_codes = [
+    #     'T' + ticket_type + ticket_variant + group_type
+    #     # E = Early Bird, R = Regular, D = on Desk
+    #     for ticket_type in ['E', 'R', 'D']
+    #     # S = Standard; L = Standard Light (no trainings); D = Day Pass
+    #     for ticket_variant in ['S', 'L', 'D']
+    #     # S = Student; P = Personal; C = Company
+    #     for group_type in ['S', 'P', 'C']
+    # ]
+
+    ticket_price = 100
+    ticket_amount = 20
+    social_event_price = 10
+    social_event_amount = 5
+
+    def create_fare(code, price, name, type, vat_rate):
+        # TODO: maybe use Fare factory(?)
+        fare = Fare.objects.create(
+            conference=settings.CONFERENCE_CONFERENCE,
+            name=name,
+            description=name,
+            code=code,
+            price=price,
+            ticket_type=type,
+            start_validity=date.today() - timedelta(days=10),
+            end_validity=date.today() + timedelta(days=10),
+        )
+
+        # fare also needs VAT
+        vat_rate, _ = Vat.objects.get_or_create(value=vat_rate)
+        VatFare.objects.get_or_create(fare=fare, vat=vat_rate)
+        return fare
+
+    create_fare("TRSP",  # Ticket Regular Standard Personal
+                ticket_price,
+                "Regular Standard Ticket",
+                "conference",
+                vat_rate=10)
+
+    create_fare("VOUPE03",  # hardcoded social event format
+                social_event_price,
+                "Social Event One",
+                "event",
+                vat_rate=10)
+
+    create_fare("S123",  # some random social event id
+                social_event_price,
+                "Social Event Two",
+                "other",
+                vat_rate=20)
+
+    # 4. If Fare is created we should have one input on the cart.
+    response = client.get(cart_url)
+    assert template_used(response, "p3/cart.html")
+    _response_content = response.content.decode('utf-8')
+
+    assert 'Sorry, no tickets are available' not in _response_content
+    assert 'Buy tickets (1 of 2)' in _response_content
+
+    # There are plenty of tds but only TRSP should have data-fare set
+    assert 'td class="fare" data-fare="TRSP">' in _response_content
+    assert 'td class="fare" data-fare="TDCP">' not in _response_content
+    assert 'td class="fare" data-fare="">' in _response_content
+    # social events
+    assert 'td class="fare" data-fare="VOUPE03">' in _response_content
+    assert 'td class="fare" data-fare="S123">' in _response_content
+
+    # and one input for TRSP where you can specify how many tickets
+    # TODO: maybe it should have a different type than text?
+    assert '<input type="text" size="2" name="TRSP"' in _response_content
+
+    # 5. Try buying some tickets
+    # FIXME: looks like the max_tickets is enforced only with javascript
+    assert ticket_amount > conference_settings.MAX_TICKETS
```

Comment:
do we actually need to limit the number of tickets?

Reply:
dunno, but but there's definitely some javascript in the cart, that basically does min(your_value, 6), and you can't buy more than six tickets at once. seems to only apply to conference tickets, not to social event tickets.

Reply:
Yes, we need to. I don't know if 6 is a sensible number, but not more than 10. We can talk about this limit elsewhere.

Reply:
@alexsavio and @artcz and @malemburg what do you think about this limit of 6 or 10? usually, the limit is 10. Is it the case when you are a sponsor, do you need to buy the tickets? if yes, we will have an issue because a sponsor can have a lot of attendees, example, facebook, microsoft, redhat, etc... they come in group ;-)